pkg/apis/extensions/types.go

```diff
@@ -226,6 +226,14 @@ type DeploymentSpec struct {
 	// Value of this key is hash of DeploymentSpec.PodTemplateSpec.
 	// No label is added if this is set to empty string.
 	UniqueLabelKey string `json:"uniqueLabelKey,omitempty"`
+
+	// The config this deployment is rolling back to. Will be cleared after rollback is done.
+	Rollback *RollbackConfig `json:"rollback,omitempty"`
```

Comment:
I like @kargakis's proposal to make this a separate subresource.

Look at how pod/binding is implemented. 

POST (Create) to /rollback should update the Deployment spec to match the specified revision. Think of it as a convenience mechanism for looking up the old revision details and updating Deployment to match.

(Note: I prefer the term "revision" vs. "version" because the latter is so overloaded in the system, and each RC is the result of a revision to the Deployment spec.)